src/expressions/match-expr.md

```diff
@@ -132,34 +77,13 @@ let message = match x {
 assert_eq!(message, "a few");
 ```
 
-Other forms of [range] \(`..` for an exclusive range, or any range with one or
-both endpoints left unspecified) are not supported in matches. The
-syntax `...` is also accepted for inclusive ranges in patterns only, for
-backwards compatibility.
-
-Range patterns only work [`char`] and [numeric types]. A range pattern may not
-be a sub-range of another range pattern inside the same `match`.
+Please notice that the `2..=9` is a [Range Pattern], not a [Range Expression]
+and, thus, only those types of ranges supported by range patterns can be used
+in match arms.
 
-Slice patterns can match both arrays of fixed size and slices of dynamic size.
-```rust
-// Fixed size
-let arr = [1, 2, 3];
-match arr {
-    [1, _, _] => "starts with one",
-    [a, b, c] => "starts with something else",
-};
-```
-```rust
-// Dynamic size
-let v = vec![1, 2, 3];
-match v[..] {
-    [a, b] => { /* this arm will not apply because the length doesn't match */ }
-    [a, b, c] => { /* this arm will apply */ }
-    _ => { /* this wildcard is required, since we don't know length statically */ }
-}
-```
+A range pattern may not be a sub-range of another range pattern inside the same `match`.
```

Comment:
This doesn't seem to be true; https://play.rust-lang.org/?gist=49813df7725fc36a71959a136841ce14&version=nightly shows only a warning.